addons/project/static/src/components/subtask_one2many_field/subtask_one2many_field.js

```diff
@@ -1,22 +1,21 @@
 import { registry } from "@web/core/registry";
 import { pick } from "@web/core/utils/objects";
-import { X2ManyField, x2ManyField } from '@web/views/fields/x2many/x2many_field';
+import { X2ManyField, x2ManyField } from "@web/views/fields/x2many/x2many_field";
 
-import { SubtaskListRenderer } from './subtask_list_renderer';
+import { SubtaskListRenderer } from "./subtask_list_renderer";
 
 export class SubtaskOne2ManyField extends X2ManyField {
     static components = {
         ...X2ManyField.components,
         ListRenderer: SubtaskListRenderer,
     };
 
-    async switchToForm(record, options) {
-        await this.props.record.save();
+    async switchToFormAction(resId, options) {
```

Comment:
what about a `getFormActionContext` method instead? It looks like it's all/only about the context.